### PR TITLE
Introduces new cvar cl_audiopaused to control if the audio loop

### PR DESCRIPTION
**Pull request by devnexen**
ought to go on or not when the game pauses.

### CODE REVIEW

File: src/client/menu/menu.c [hunk 1]
```diff
@@ -2365,6 +2373,13 @@ Options_MenuInit(void)
     s_options_quality_list.generic.callback = UpdateSoundQualityFunc;
     s_options_quality_list.itemnames = quality_items;
 
+    s_options_audiopause_box.generic.type = MTYPE_SPINCONTROL;
+    s_options_audiopause_box.generic.x = 0;
+    s_options_audiopause_box.generic.y = (y += 10);
+    s_options_audiopause_box.generic.name = "pause sound on pause";
```
Comment (Yamagi) [inline src/client/menu/menu.c:2379]: This overflows the menu on 320x280. And do we really want to use the last available line in the options menu for a rather obscure feature like this? I think having it as cvar only is good enough.

File: src/client/sound/openal.c [hunk 2]
```diff
@@ -854,11 +854,12 @@ AL_AddLoopSounds(void)
 	int num;
 	entity_state_t *ent;
 
-	if ((cls.state != ca_active) || cl_paused->value || !s_ambient->value)
+	if ((cls.state != ca_active) || (cl_paused->value && cl_audiopaused->value) || !s_ambient->value)
 	{
 		return;
 	}
 
+	memset(&sounds, 0, sizeof(int) * MAX_EDICTS);
```
Comment (Yamagi) [inline src/client/sound/openal.c:862]: Why is this necessary?

**Review by Yamagi — CHANGES_REQUESTED** [review]
I have added some comments.

File: src/client/sound/openal.c (hunk 2 above)
Comment (devnexen) [reply src/client/sound/openal.c:862]: for matching the sdl backend mostly.

**Comment by Yamagi** [comment]
Thank you :)

**Pull request merged**